src/lib/auth/credentials.mjs

```diff
@@ -2,27 +2,50 @@ import { asValue, Lifetime } from "awilix";
 
 import { container } from "../../cli.mjs";
 import { FaunaAccountClient } from "../fauna-account-client.mjs";
-import { cleanupSecretsFile } from "../file-util.mjs";
 import { AccountKeys } from "./accountKeys.mjs";
 import { DatabaseKeys } from "./databaseKeys.mjs";
 
 const validateCredentialArgs = (argv) => {
+  const logger = container.resolve("logger");
   if (argv.database && argv.secret) {
     throw new Error(
-      "Cannot provide both a database and a secret. Please provide one or the other.",
+      "Cannot use both the '--secret' and '--database' options together. Please specify only one.",
+    );
+  } else if (argv.role && argv.secret) {
+    // The '--role' option is not supported when using a secret. Secrets have an
+    // implicit role.
+    logger.warn(
+      "The '--role' option is not supported when using a secret. It will be ignored.",
     );
   }
 };
 
 export class Credentials {
   constructor(argv) {
     // Get rid of orphaned database keys in the local storage
-    cleanupSecretsFile();
     // Make sure auth-related arguments from users are legal
     validateCredentialArgs(argv);
-
     this.accountKeys = new AccountKeys(argv);
     this.databaseKeys = new DatabaseKeys(argv, this.accountKeys.key);
+    this.cleanupSecretsFile();
+  }
+
+  /**
+   * Steps through account keys in local filesystem and if they are not found in the secrets file,
+   *   delete the stale entries on the secrets file.
+   */
+  cleanupSecretsFile() {
+    const accountKeyData = this.accountKeys.keyStore.getFile();
+    const accountKeys = Object.values(accountKeyData).map(
+      (value) => value.accountKey,
+    );
+    const secretKeyData = this.databaseKeys.keyStore.getFile();
+    Object.keys(secretKeyData).forEach((accountKey) => {
+      if (!accountKeys.includes(accountKey)) {
+        this.databaseKeys.keyStore.updateAccountKey(accountKey);
+        this.databaseKeys.keyStore.deleteAllDBKeysForAccount();
+      }
+    });
```

Comment:
Moved the cleanup to this class because
previously I was creating new AccountKeyStorage and DatabaseKeyStorage here which is unnecessary if I just reuse what's already been made.